ocs_ingester/utils/metrics.py

```diff
@@ -14,12 +15,32 @@ def wrapper(self, *args, **kwargs):
             # of when settings are changed at runtime is when the ingester command line entrypoint is used.
             @metric_timer_with_tags(
                 metric_name=metric_name,
-                asynchronous=settings.SUBMIT_METRICS_ASYNCHRONOUSLY,
-                **settings.EXTRA_METRICS_TAGS
+                asynchronous=ingester_settings.SUBMIT_METRICS_ASYNCHRONOUSLY,
+                **ingester_settings.EXTRA_METRICS_TAGS
             )
             @functools.wraps(method)
             def run_method(self, *args, **kwargs):
                 return method(self, *args, **kwargs)
             return run_method(self, *args, **kwargs)
         return wrapper
     return method_timer_decorator
+
+@method_timer('ingester.upload_file')
+def upload_and_collect_metrics(filestore_class, datafile):
+    start_time = datetime.utcnow()
+    # Upload the file to s3 and get version information back
+    version = filestore_class.store_file(data_file=datafile)
+    # Record metric for the bytes transferred / time to upload
+    upload_time = datetime.utcnow() - start_time
+    bytes_per_second = len(datafile.open_file) / upload_time.total_seconds()
+    send_tsdb_metric(
+        metric_name='ingester.s3_upload_bytes_per_second',
```

Comment:
Does this only apply to s3?

Reply:
no, but I didn't want to change the metric name since we already have been using it.